Wrappers/Python/cil/plugins/tigre/ProjectionOperator.py

```diff
@@ -24,22 +26,78 @@
+        "Please install it via conda as tigre from the ccpi channel")
 
 try:
     from tigre.utilities.gpu import GpuIds
     has_gpu_sel = True
 except ModuleNotFoundError:
     has_gpu_sel = False
 
+
 class ProjectionOperator(LinearOperator):
+    """
+        ProjectionOperator configures and calls TIGRE Projectors for your dataset.
+
+        Please refer to the TIGRE documentation for futher descriptions
+        https://github.com/CERN/TIGRE
+        https://iopscience.iop.org/article/10.1088/2057-1976/2/5/055010
+                        
+
+        Parameters
+        ----------
+
+        image_geometry : `ImageGeometry`, default used if None
+            A description of the area/volume to reconstruct
+
+        acquisition_geometry :`AcquisitionGeometry`, `BlockGeometry`
+            A description of the acquisition data. If passed a BlockGeometry it will return a BlockOperator.
+
+        direct_method : str,  default 'interpolated'
+            The method used by the forward projector, 'Siddon' for ray-voxel intersection, 'interpolated' for interpolated projection
+
+        adjoint_weights : str, default 'matched'
+            The weighting method used by the cone-beam backward projector, 'matched' for weights to approximately match the 'interpolated' forward projector, 'FDK' for FDK weights
+
+        Example
+        -------
+        >>> from cil.plugins.tigre import ProjectionOperator
+        >>> PO = ProjectionOperator(image.geometry, data.geometry)
+        >>> forward_projection = PO.direct(image)
+        >>> backward_projection = PO.adjoint(data)
+
+        """
+    def __new__(cls, image_geometry=None, acquisition_geometry=None, \
+        direct_method='interpolated',adjoint_weights='matched', **kwargs):
+        if isinstance(acquisition_geometry, BlockGeometry):
+            logging.info("BlockOperator is returned.")
+
+            K = []
+            for ag in acquisition_geometry:
+                K.append(
+                    ProjectionOperator_ag(image_geometry=image_geometry, acquisition_geometry=ag, \
+                        direct_method=direct_method, adjoint_weights=adjoint_weights, **kwargs)
+                )
+            return BlockOperator(*K)
+        else:
+            logging.info("Standard Operator is returned.")
+            return super(ProjectionOperator,
+                         cls).__new__(ProjectionOperator_ag)
+
+
+class ProjectionOperator_ag(ProjectionOperator):
     '''TIGRE Projection Operator'''
 
-    def __init__(self, image_geometry=None, acquisition_geometry=None, direct_method='interpolated',adjoint_weights='matched', **kwargs): 
-    
+    def __init__(self,
+                 image_geometry=None,
+                 acquisition_geometry=None,
+                 direct_method='interpolated',
+                 adjoint_weights='matched',
+                 **kwargs):
         """
         ProjectionOperator configures and calls TIGRE Projectors for your dataset.
 
```

Comment:
We need to check the docs, but I assume we should no longer expose this class

Reply:
It's actually not exposed, you cannot reach it unless you know it's there: it's not in the `__init__.py` and it's not in the documentation.